com.unity.ml-agents/Runtime/Academy.cs

```diff
@@ -414,7 +414,9 @@ void InitializeEnvironment()
             EnableAutomaticStepping();
 
             SideChannelManager.RegisterSideChannel(new EngineConfigurationChannel());
+#if MLA_UNITY_ANALYTICS_MODULE_ENABLED
```

Comment:
I think as a rule of thumb, the defines should only be used in the `Unity.MLAgents.Analytics` namespace. Using the defines to make the public `Analytics` methods no-ops means the user doesn't need to know about whether or not it's enabled; helps shrink the effective interface. 

In particular, there's a bug here because the capabilities still say that training analytics are enabled, but you're not registering the SideChannel, so unrecognized message UUIDs will be received.

Reply:
done